monai/losses/unified_focal_loss.py

```
@@ -212,12 +248,11 @@ def forward(self, y_pred: torch.Tensor, y_true: torch.Tensor) -> torch.Tensor:
 
-        if y_pred.shape[1] == 1:
-            y_pred = one_hot(y_pred, num_classes=self.num_classes)
-            y_true = one_hot(y_true, num_classes=self.num_classes)
-
-        if torch.max(y_true) != self.num_classes - 1:
-            raise ValueError(f"Please make sure the number of classes is {self.num_classes-1}")
+        if y_true.shape[1] != self.num_classes or torch.max(y_true) > self.num_classes - 1:
+            raise ValueError(
+                f"y_true must have {self.num_classes} channels (one-hot) or label values in [0, {self.num_classes - 1}], "
+                f"but got shape {y_true.shape} with max value {torch.max(y_true)}"
+            )
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)